// wdio-wait-for/dist/cjs/element.js
"use strict";
const importElementPromise = import('../element/index.js');
exports.elementToBeClickable = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).elementToBeClickable;
        return fn(selectorOrElement).call(this);
    };
};
exports.elementToBeEnabled = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).elementToBeEnabled;
        return fn(selectorOrElement).call(this);
    };
};
exports.elementToBeSelected = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).elementToBeSelected;
        return fn(selectorOrElement).call(this);
    };
};
exports.invisibilityOf = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).invisibilityOf;
        return fn(selectorOrElement).call(this);
    };
};
exports.numberOfElementsToBe = function (selectorOrElementArray, expectedNumber) {
    return async function () {
        const fn = (await importElementPromise).numberOfElementsToBe;
        return fn(selectorOrElementArray, expectedNumber).call(this);
    };
};
exports.numberOfElementsToBeLessThan = function (selectorOrElementArray, expectedNumber) {
    return async function () {
        const fn = (await importElementPromise).numberOfElementsToBeLessThan;
        return fn(selectorOrElementArray, expectedNumber).call(this);
    };
};
exports.numberOfElementsToBeMoreThan = function (selectorOrElementArray, expectedNumber) {
    return async function () {
        const fn = (await importElementPromise).numberOfElementsToBeMoreThan;
        return fn(selectorOrElementArray, expectedNumber).call(this);
    };
};
exports.presenceOf = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).presenceOf;
        return fn(selectorOrElement).call(this);
    };
};
exports.sizeOfElementsToBe = function (selectorOrElement, expectedSize) {
    return async function () {
        const fn = (await importElementPromise).sizeOfElementsToBe;
        return fn(selectorOrElement, expectedSize).call(this);
    };
};
exports.stalenessOf = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).stalenessOf;
        return fn(selectorOrElement).call(this);
    };
};
exports.textToBePresentInElement = function (selectorOrElement, expectedText) {
    return async function () {
        const fn = (await importElementPromise).textToBePresentInElement;
        return fn(selectorOrElement, expectedText).call(this);
    };
};
exports.textToBePresentInElementValue = function (selectorOrElement, expectedValue) {
    return async function () {
        const fn = (await importElementPromise).textToBePresentInElementValue;
        return fn(selectorOrElement, expectedValue).call(this);
    };
};
exports.visibilityOf = function (selectorOrElement) {
    return async function () {
        const fn = (await importElementPromise).visibilityOf;
        return fn(selectorOrElement).call(this);
    };
};
